src/app/server/Server.cpp

```diff
@@ -72,330 +66,74 @@ constexpr bool isRendezvousBypassed()
-        }
-
-        if (mDelegate != nullptr)
-        {
-            mDelegate->OnPairingWindowClosed();
-        }
-
-        return CHIP_NO_ERROR;
-    }
-
-    void SetDelegate(AppDelegate * delegate) { mDelegate = delegate; }
-    void SetBLE(bool ble) { isBLE = ble; }
-    void SetFabricIndex(FabricIndex id) { mFabric = id; }
-
-private:
-    AppDelegate * mDelegate = nullptr;
-    FabricIndex mFabric;
-    bool isBLE = true;
-};
-
-DemoTransportMgr gTransports;
-SecureSessionMgr gSessions;
-RendezvousServer gRendezvousServer;
-CASEServer gCASEServer;
-Messaging::ExchangeManager gExchangeMgr;
-ServerRendezvousAdvertisementDelegate gAdvDelegate;
-
-class ServerCallback : public ExchangeDelegate
-{
-public:
-    CHIP_ERROR OnMessageReceived(Messaging::ExchangeContext * exchangeContext, const PacketHeader & packetHeader,
-                                 const PayloadHeader & payloadHeader, System::PacketBufferHandle && buffer) override
-    {
-        CHIP_ERROR err = CHIP_NO_ERROR;
-        // as soon as a client connects, assume it is connected
-        VerifyOrExit(!buffer.IsNull(), ChipLogError(AppServer, "Received data but couldn't process it..."));
-        VerifyOrExit(mSessionMgr != nullptr, ChipLogError(AppServer, "SecureSessionMgr is not initilized yet"));
-        HandleDataModelMessage(exchangeContext, std::move(buffer));
-
-    exit:
-        return err;
-    }
-
-    void OnResponseTimeout(ExchangeContext * ec) override
-    {
-        ChipLogProgress(AppServer, "Failed to receive response");
-        if (mDelegate != nullptr)
-        {
-            mDelegate->OnReceiveError();
-        }
-    }
-
-    void SetDelegate(AppDelegate * delegate) { mDelegate = delegate; }
-    void SetSessionMgr(SecureSessionMgr * mgr) { mSessionMgr = mgr; }
-
-private:
-    AppDelegate * mDelegate        = nullptr;
-    SecureSessionMgr * mSessionMgr = nullptr;
-};
-
-secure_channel::MessageCounterManager gMessageCounterManager;
-ServerCallback gCallbacks;
-SecurePairingUsingTestSecret gTestPairing;
-
-#if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY_CLIENT
-
-chip::Protocols::UserDirectedCommissioning::UserDirectedCommissioningClient gUDCClient;
-
-#endif // CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY_CLIENT
-
-void HandlePairingWindowTimeout(System::Layer * aSystemLayer, void * aAppState)
-{
-    ClosePairingWindow();
-}
-
 } // namespace
 
-CHIP_ERROR OpenBasicCommissioningWindow(ResetFabrics resetFabrics, uint16_t commissioningTimeoutSeconds,
-                                        chip::PairingWindowAdvertisement advertisementMode)
-{
-    // TODO(cecille): If this is re-called when the window is already open, what should happen?
-    gDeviceDiscriminatorCache.RestoreDiscriminator();
-
-    uint32_t pinCode;
-    ReturnErrorOnFailure(DeviceLayer::ConfigurationMgr().GetSetupPinCode(pinCode));
+namespace chip {
 
-    RendezvousParameters params;
+Server Server::sServer;
```

Comment:
What does this statement do? Does this actually construct the Server singleton? Does that not already happen in Server.h?

Reply:
That's only declaration and this actually constructs the singleton.